run.go

```diff
@@ -346,18 +400,64 @@ func (d *InMAP) checkOutputNames(n ...string) error {
 // layers, otherwise only the ground-level layer is returned.
 // outputVariables is a list of the names of the variables for which data should be
 // returned.
-func (d *InMAP) Results(allLayers bool, outputVariables ...string) (map[string][]float64, error) {
-	if err := d.checkOutputNames(outputVariables...); err != nil {
+func (d *InMAP) Results(allLayers bool, outputVariables map[string]string) (map[string][]float64, error) {
+
+	functions := map[string]govaluate.ExpressionFunction{
+		"exp": func(arg ...interface{}) (interface{}, error) {
+			return (float64)(math.Exp(arg[0].(float64))), nil
```

Comment:
if len(arg) != 1 {
return nil, fmt.Errorf("inmap:  got %d arguments for function 'exp', but need 1", len(arg))
}

Also add statements like this for the other functions to make sure they are getting the correct number of arguments.